Remove obsolete files

sciascid opened the pull request:
* sql/wsrep_tc.h - empty file, which was committed in by mistake.
* sql/wsrep_hton.cc - wsrep handlerton is no longer used by galera 4.

SachinSetiya commented on sql/wsrep_hton.cc:
```diff
@@ -1,659 +0,0 @@
-/*
-  Cleanup after local transaction commit/rollback, replay or TOI.
-*/
-void wsrep_cleanup_transaction(THD *thd)
```
wsrep_cleanup_transaction still present in wsrep_mysqld.h

SachinSetiya commented on sql/wsrep_hton.cc:
```diff
@@ -1,659 +0,0 @@
-  Calls wsrep->post_commit() for locally executed transactions that have
-  got seqno from provider (must commit) and don't require replaying.
- */
-void wsrep_post_commit(THD* thd, bool all)
```
it is called in sql_load.cc

SachinSetiya commented on sql/wsrep_hton.cc:
still present in sql_load.cc

sciascid commented:
@SachinSetiya Thanks for your review. I addressed your comments in a follow up commit.

About the comments in wsrep_mysqld.h: I removed most of them. But turns out that some are needed to make mariadb compile WITH_WSREP=OFF. Those I restored.

SachinSetiya commented:
@sciascid , Thanks for addressing comments, It compiles with wsrep_on and off/
